### PR TITLE
Locus list ui

### PR DESCRIPTION
**Pull request by hanars**
still needs tests but otherwise this is good to go

### CODE REVIEW

File: seqr/model_utils.py [hunk 1]
```diff
@@ -210,6 +213,9 @@ def convert_seqr_kwargs_to_xbrowse_kwargs(seqr_model, **kwargs):
     if seqr_class_name == "Individual" and "family" in xbrowse_kwargs:
         xbrowse_kwargs["project"] = getattr(seqr_model, "family").project
 
+    if seqr_class_name == "LocusList" and 'name' in xbrowse_kwargs:
+        xbrowse_kwargs['slug'] = xbrowse_kwargs['name'].replace(' ', '-').lower()
```
Comment (bw2) [inline seqr/model_utils.py]: might be better to call `slugify` method instead. (Other places in the code use it)

File: ui/shared/components/panel/genes/LocusListDetail.jsx [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+
+const mapStateToProps = (state, ownProps) => ({
+  locusList: getLocusListsByGuid(state)[ownProps.locusListGuid] || {},
+})
+
+const mapDispatchToProps = {
+  onSubmit: updateLocusList,
+}
+
+export { LocusListDetail }
```
Comment (bw2) [inline ui/shared/components/panel/genes/LocusListDetail.jsx]: Should this be export {  LoadedLocusListDetail }?

Comment (hanars) [reply ui/shared/components/panel/genes/LocusListDetail.jsx]: no, thats exported as the default on the next line, this is intentionally exporting the unconnected component for testing

File: seqr/model_utils.py (hunk 1 above)
Comment (bw2) [reply seqr/model_utils.py]: or django.utils.text.slugify

File: ui/shared/components/form/ReduxFormWrapper.jsx [hunk 3]
```diff
@@ -116,35 +146,25 @@ class ReduxFormWrapper extends React.Component {
     } else if (this.props.submitFailed) {
       saveStatus = RequestStatus.ERROR
     }
+
+    // redux-form does not support throwing submission warnings so this is a work around
+    const submissionWarnings = this.props.warning || (this.props.error && this.props.error.map(error => error.warning).filter(warning => warning))
```
Comment (bw2) [inline ui/shared/components/form/ReduxFormWrapper.jsx]: I was initially in favor of warnings but in retrospect there's no good way to fit them into a dialog like this, so would vote for getting rid of them - the error message when ignoreInvalidItems=false seems sufficient, and then the form can just close on submit when ignoreInvalidItems=true.

File: seqr/views/apis/locus_list_api.py [hunk 4]
```diff
@@ -0,0 +1,233 @@
...
+                ))
+
+    gene_symbols_to_ids = get_gene_symbols_to_gene_ids(new_gene_symbols)
+    invalid_items += [symbol for symbol, gene_id in gene_symbols_to_ids.items() if not gene_id]
+    new_genes = get_genes([gene_id for gene_id in gene_symbols_to_ids.values() if gene_id] + list(new_gene_ids))
+    invalid_items += [gene_id for gene_id, gene in new_genes.items() if not gene]
+    new_genes = {gene_id: gene for gene_id, gene in new_genes.items() if gene}
+
+    if invalid_items and not request_json.get('ignoreInvalidItems'):
+        return None, None, invalid_items, 'This list contains invalid genes/ intervals. Update them, or select the "Ignore invalid genes and intervals" checkbox to ignore.'
```
Comment (bw2) [inline seqr/views/apis/locus_list_api.py]: could the invalid items be appended to this error message, so the error box shows one message instead of 2?

**Pull request merged**